parser/encoder.go

```diff
@@ -79,9 +79,9 @@ func (e *encoder) encodeAsString(packet *Packet) types.BufferInterface {
 	}
 	// json data
 	if nil != packet.Data {
-		if pds, is := packet.Data.(string); is {
+		if packet.preSerializedData != "" {
```

Comment:
I need to run more test cases to ensure that this request is OK. It may take longer.

Reply:
There's no rush.  I've already taken the encoder from this branch and I'm using it in my project that way via the config.Encoder.